### PR TITLE
(service-client): Add constructors in job client to accept aad and sa… …

### PR DESCRIPTION
**Pull request by vinagesh**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

### CODE REVIEW

File: iothub/service/src/HttpRegistryManager.cs [hunk 1]
```diff
@@ -54,12 +54,12 @@ internal class HttpRegistryManager : RegistryManager
         private IHttpClientHelper _httpClientHelper;
         private readonly string _iotHubName;
 
-        internal HttpRegistryManager(IotHubConnectionString connectionString, HttpTransportSettings transportSettings)
+        internal HttpRegistryManager(IotHubConnectionProperties connectionProperties, HttpTransportSettings transportSettings)
```
Comment (vinagesh) [inline iothub/service/src/HttpRegistryManager.cs:57]: All these properties are now in a base class, making the HttpRegistryManager independent of the auth type.

**Review by timtay-microsoft — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/service/src/JobClient/JobClient.cs [hunk 2]
```diff
@@ -41,6 +47,66 @@ public static JobClient CreateFromConnectionString(string connectionString, Http
...
+        /// <param name="transportSettings">The HTTP transport settings.</param>
+        /// <returns>An instance of <see cref="JobClient"/>.</returns>
+        public static JobClient Create(
+            string hostName,
+            TokenCredential credential,
+            HttpTransportSettings transportSettings = default)
+        {
+            if (string.IsNullOrEmpty(hostName))
+            {
+                throw new ArgumentNullException($"{nameof(hostName)} is null or empty");
```
Comment (drwill-ms) [inline iothub/service/src/JobClient/JobClient.cs:66]: same as other PR, use the overload that takes the param name and a description

File: iothub/service/src/JobClient/JobClient.cs [hunk 3]
```diff
@@ -41,6 +47,66 @@ public static JobClient CreateFromConnectionString(string connectionString, Http
...
+            }
+
+            if (credential == null)
+            {
+                throw new ArgumentNullException($"{nameof(credential)} is null");
+            }
+
+            TlsVersions.Instance.SetLegacyAcceptableVersions();
```
Comment (drwill-ms) [inline iothub/service/src/JobClient/JobClient.cs:74]: same, why call this?

File: iothub/service/src/JobClient/JobClient.cs [hunk 4]
```diff
@@ -41,6 +47,66 @@ public static JobClient CreateFromConnectionString(string connectionString, Http
...
+
+            IotHubTokenCrendentialProperties tokenCredentialProperties = new IotHubTokenCrendentialProperties(hostName, credential);
```
Comment (drwill-ms) [inline iothub/service/src/JobClient/JobClient.cs:76]: var

File: iothub/service/src/JobClient/JobClient.cs [hunk 5]
```diff
@@ -41,6 +47,66 @@ public static JobClient CreateFromConnectionString(string connectionString, Http
...
+            return new HttpJobClient(tokenCredentialProperties, transportSettings ?? new HttpTransportSettings());
+        }
+
+        /// <summary>
+        /// Creates an instance of <see cref="JobClient"/>.
+        /// </summary>
+        /// <param name="hostName">IoT hub host name.</param>
+        /// <param name="credential">Credential that generates a SAS token to authenticate with IoT hub. See <see cref="IotHubSasCredential"/></param>
```
Comment (drwill-ms) [inline iothub/service/src/JobClient/JobClient.cs:84]: End with a period

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/src/JobClient/JobClient.cs (hunk 3 above)
Comment (vinagesh) [reply iothub/service/src/JobClient/JobClient.cs:74]: Will remove. Did not see the implementation was only for NET451

**Pull request merged**